src/System.CommandLine.Subsystems/ValueSources/AggregateValueSource.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.CommandLine.ValueSources;
+
+public class AggregateValueSource<T> : ValueSource<T>
```

Comment:
IMO all these ValueSource impls should be internal, or at the very least sealed with internal constructors.

Given there's the static API to create these, we should avoid a lack of clarity about which of two methods to use to create them.  And we shouldn't allow subclassing them without a concrete extensibility use case.